protocols/rendezvous/src/handler.rs

```diff
@@ -0,0 +1,520 @@
+                inbound,
+                SubstreamState::Active(Outbound::PendingOpen(Message::Unregister(namespace))),
+            ),
+            (
+                InEvent::DiscoverRequest {
+                    namespace,
+                    cookie,
+                    limit,
+                },
+                inbound,
+                SubstreamState::None,
+            ) => (
+                inbound,
+                SubstreamState::Active(Outbound::PendingOpen(Message::Discover {
+                    namespace,
+                    cookie,
+                    limit,
+                })),
+            ),
+            (
+                InEvent::RegisterResponse { ttl },
+                SubstreamState::Active(Inbound::PendingBehaviour(substream)),
+                outbound,
+            ) => (
+                SubstreamState::Active(Inbound::PendingSend(
+                    substream,
+                    Message::RegisterResponse(Ok(ttl)),
+                )),
+                outbound,
+            ),
+            (
+                InEvent::DeclineRegisterRequest(error),
+                SubstreamState::Active(Inbound::PendingBehaviour(substream)),
+                outbound,
+            ) => (
+                SubstreamState::Active(Inbound::PendingSend(
+                    substream,
+                    Message::RegisterResponse(Err(error)),
+                )),
+                outbound,
+            ),
+            (
+                InEvent::DiscoverResponse { discovered, cookie },
+                SubstreamState::Active(Inbound::PendingBehaviour(substream)),
+                outbound,
+            ) => (
+                SubstreamState::Active(Inbound::PendingSend(
+                    substream,
+                    Message::DiscoverResponse(Ok((discovered, cookie))),
+                )),
+                outbound,
+            ),
+            (
+                InEvent::DeclineDiscoverRequest(error),
+                SubstreamState::Active(Inbound::PendingBehaviour(substream)),
+                outbound,
+            ) => (
+                SubstreamState::Active(Inbound::PendingSend(
+                    substream,
+                    Message::DiscoverResponse(Err(error)),
+                )),
+                outbound,
+            ),
+            (event, inbound, outbound) => {
+                log::warn!(
+                    "Neither {:?} nor {:?} can handle {:?}",
+                    inbound,
+                    outbound,
+                    event
+                );
+
+                (inbound, outbound)
+            }
+        };
+
+        self.inbound = inbound;
+        self.outbound = outbound;
+    }
+
+    fn inject_dial_upgrade_error(
+        &mut self,
+        _: Self::OutboundOpenInfo,
+        _: ProtocolsHandlerUpgrErr<Void>,
+    ) {
+    }
+
+    fn connection_keep_alive(&self) -> KeepAlive {
+        KeepAlive::Yes
```

Comment:
To keep things simple, would the following work for now?

On each activity, e.g. inbound or outbound request, set `self.keep_alive = KeepAlive::Until(Instant::now() + ten_seconds)`. In `connection_keep_alive` return `self.keep_alive`.